modules/gapi/src/backends/fluid/gfluidimgproc.cpp

```diff
@@ -800,10 +808,28 @@ GAPI_FLUID_KERNEL(GFluidGaussBlur, cv::gapi::imgproc::GGaussBlur, true)
         int chan  = src.meta().chan;
         int length = width * chan;
 
-        float *buf[3];
+        const int buffSize = 5;
+        float *buf[buffSize];
+
+        int buffLen = 0;
+        if (ksize.height >= buffSize)
+            buffLen = buffSize;
+        else
+            buffLen = ksize.height;
+
         buf[0] = ky + kysize;
-        buf[1] = buf[0] + length;
-        buf[2] = buf[1] + length;
+        for (int i = 1; i < buffLen; ++i)
+        {
+            buf[i] = buf[i-1] + length;
+        }
```

Comment:
I believe this changes are coming from fix #15874 right? Let's wait for that PR to be merged so than this one would only contain optimization-relevant changes